Python/compile.c

```diff
@@ -285,9 +287,16 @@ compiler_init(struct compiler *c)
 {
     memset(c, 0, sizeof(struct compiler));
 
+    c->c_const_cache = PyDict_New();
+    if (!c->c_const_cache) {
+        return 0;
+    }
+
     c->c_stack = PyList_New(0);
-    if (!c->c_stack)
+    if (!c->c_stack) {
+        Py_DECREF(c->c_const_cache);
```

Comment:
Use maybe Py_CLEAR() to keep c consistent (otherwise, compiler_free() may crash).